### PR TITLE
XR_KHR_ANDROID_CREATE_INSTANCE_EXTENSION_NAME missing, required for Monado but still works on Quest

### PR DESCRIPTION
**Pull request by BattleAxeVR**
See subject line: without the XR_KHR_ANDROID_CREATE_INSTANCE_EXTENSION_NAME instance extension added, and the associated struct initialized and added to the .next of XrInstanceCreateInfo, the app fails to load on Monado runtimes.

image

There are probably better or more elegant ways of getting the activity and vm variables passed into this struct in a "clean" way, but this is the best I could do just to get up and running. I don't care if this particular PR gets merged or not, just that the issue is fixed one way or another so my own codebase doesn't diverge significantly, making future merges to my branch more difficult and time-consuming (conflicts).

Reason: I'd like my IGL build to run on all XR devices I own, not just Quest. Being vendor-agnostic is the whole reason OpenXR exists in the first-place, after all.

Btw this XrInstanceCreateInfoAndroidKHR struct is also used in the hello_xr sample from OpenXR SDK Source repo here:

https://github.com/KhronosGroup/OpenXR-SDK-Source/blob/455583cc6ddb463f7433fc0e9bc2a4b9b0c943c6/src/tests/hello_xr/openxr_program.cpp#L190

### CODE REVIEW

**Comment by BattleAxeVR** [comment]
There's a linux build error, I don't have Linux installed here so I couldn't test it. But anyway it's not in the code I touched so not my responsibility to fix this.

image

**Comment by corporateshark** [comment]
>There's a linux build error, removing the STD prefix and see if it passes

We will make a fix for that. For now you can just `#include <cmath>`.

**Comment by corporateshark** [comment]
Please sign that CLA so I can import your PR.

File: shell/openxr/mobile/XrApp.cpp [hunk 1]
```diff
@@ -569,7 +577,7 @@ void XrApp::handleXrEvents() {
 
 void XrApp::handleSessionStateChanges(XrSessionState state) {
   if (state == XR_SESSION_STATE_READY) {
-    assert(resumed_);
+    //assert(resumed_);
```
Comment (corporateshark) [inline shell/openxr/mobile/XrApp.cpp:580]:
```
#if !defined(IGL_CMAKE_BUILD)
  assert(resumed_);
#endif // IGL_CMAKE_BUILD
```

File: shell/openxr/mobile/opengl/XrAppImplGLES.cpp [hunk 2]
```diff
@@ -20,9 +20,14 @@ std::vector<const char*> XrAppImplGLES::getXrRequiredExtensions() const {
   return {
       XR_KHR_OPENGL_ES_ENABLE_EXTENSION_NAME,
       XR_KHR_COMPOSITION_LAYER_DEPTH_EXTENSION_NAME,
+      XR_KHR_ANDROID_CREATE_INSTANCE_EXTENSION_NAME
```
Comment (corporateshark) [inline shell/openxr/mobile/opengl/XrAppImplGLES.cpp:23]: Please wrap in `#if defined(IGL_CMAKE_BUILD)`.

File: shell/openxr/mobile/opengl/XrAppImplGLES.cpp [hunk 3]
```diff
@@ -20,9 +20,14 @@ std::vector<const char*> XrAppImplGLES::getXrRequiredExtensions() const {
...
   };
 }
 
+void* XrAppImplGLES::getInstanceCreateExtension() {
+    return &instanceCreateInfoAndroid_;
```
Comment (corporateshark) [inline shell/openxr/mobile/opengl/XrAppImplGLES.cpp:28]:
```
#if defined(IGL_CMAKE_BUILD)
  return &instanceCreateInfoAndroid_;
#else
  return nullptr;
#endif // IGL_CMAKE_BUILD
```

File: shell/openxr/mobile/vulkan/XrAppImplVulkan.cpp [hunk 4]
```diff
@@ -23,9 +23,14 @@ std::vector<const char*> XrAppImplVulkan::getXrRequiredExtensions() const {
       XR_KHR_VULKAN_ENABLE_EXTENSION_NAME,
       XR_FB_SWAPCHAIN_UPDATE_STATE_VULKAN_EXTENSION_NAME,
       XR_KHR_COMPOSITION_LAYER_DEPTH_EXTENSION_NAME,
+      XR_KHR_ANDROID_CREATE_INSTANCE_EXTENSION_NAME
   };
 }
 
+void* XrAppImplVulkan::getInstanceCreateExtension() {
+    return &instanceCreateInfoAndroid_;
```
Comment (corporateshark) [inline shell/openxr/mobile/vulkan/XrAppImplVulkan.cpp:31]:
```
#if defined(IGL_CMAKE_BUILD)
  return &instanceCreateInfoAndroid_;
#else
  return nullptr;
#endif // IGL_CMAKE_BUILD
```

File: shell/renderSessions/GPUStressSession.cpp [hunk 5]
```diff
@@ -486,7 +486,7 @@ void getOffset(int counter, float& x, float& y, float& z) {
     z *= counter / 2.f;
     return;
   }
-  const float grid = std::ceilf(std::powf(kCubeCount, 1.0f / 3.0f));
+  const float grid = std::ceilf(powf(kCubeCount, 1.0f / 3.0f));
```
Comment (corporateshark) [inline shell/renderSessions/GPUStressSession.cpp:489]: Adding `#include <cmath>` should fix it.

**Review by corporateshark — CHANGES_REQUESTED** [review]

**Comment by BattleAxeVR** [comment]
> Please sign that CLA so I can import your PR.

Done. It's signed.

**Pull request closed without merging**